### DISCUSSION TITLE
About lowering amp to L-size motor to prevent choking of M-size motor

### DISCUSSION THREAD

**Discussion by garfield8625** [pybricks/support#654] [Q&A] [answered]
Hello wise-ones,

I have the following issue:     While steering is engaged/actively used it chokes the driving motor
Question:     Is there a way to lower the amp that the Large motor gets via code to make it leave more juice for the M-sized?

Lego ECTO-1 motorized: 
- hub: 88009
- steering: 1 Large motor (88013), 
- drive: 1 Medium linear motor (45303)
(Medium motor is being used due to lack of space ofr bigger motor.)

# Code (if needed): 
```python
initial_drive_speed = 100

steer = Motor(Port.B, Direction.COUNTERCLOCKWISE)
back = DCMotor(Port.A, Direction.COUNTERCLOCKWISE)

remote = Remote()

old_kp, old_ki, old_kd, _, _ = steer.control.pid()

steer.control.pid(kp=old_kp*4, kd=old_kd*0.4)

left_end = steer.run_until_stalled(-200, then=Stop.HOLD)
right_end = steer.run_until_stalled(200, then=Stop.HOLD)

steer.reset_angle((right_end - left_end)/2)
steer.run_target(speed=200, target_angle=0, wait=False)

steer_angle = (((right_end - left_end)/2)-5)
print('steer angle:',steer_angle)

while True:
    pressed = remote.buttons.pressed()

    if Button.RIGHT_PLUS in pressed:
        steer.run_target(1400, -steer_angle, Stop.HOLD, False)
    elif Button.RIGHT_MINUS in pressed:
        steer.run_target(1400, steer_angle, Stop.HOLD, False)
    else:
        steer.track_target(0)

    drive_speed = 0
    if Button.LEFT_PLUS in pressed:
        drive_speed = initial_drive_speed
    if Button.LEFT_MINUS in pressed:
        drive_speed = -initial_drive_speed

    back.dc(drive_speed)

    wait(10)

```

**Comment by dlech** [comment, marked as answer]
There is a maximum voltage setting for motors: https://docs.pybricks.com/en/stable/pupdevices/motor.html#pybricks.pupdevices.Motor.settings

There is also a maximum torque setting for controlled actuation: https://docs.pybricks.com/en/stable/pupdevices/motor.html#pybricks.pupdevices.Motor.Motor.control.limits

**Reply by garfield8625** [reply]
Would you be so kind to assist regarding, how would that look like in my example code?
many thanks!

**Reply by dlech** [reply]
```python
steer.control.limits(torque=100)
```

You can `print(steer.control.limits()[2])` to see what the default setting is.

**Reply by dlech** [reply]
Or do like you did with the PID.

```python
_, _, old_torque = steer.control.limits()
steer.control.limits(torque=old_torque*0.5)
```

**Reply by garfield8625** [reply]
How would I link this command to a specific motor? - B-port in my case
or this command "steer,control.limit" already linked it to the streering motor?
I'm asking this much as im not very well versed in coding.. apologies!

**Reply by dlech** [reply]
> ```python
> steer = Motor(Port.B, Direction.COUNTERCLOCKWISE)
> ```

`steer` is already defined as the motor on port B. So `steer.anything` will act on that motor (with `anything` being literally anything listed in the docs as attributes of the motor class).

**Reply by garfield8625** [reply]
MANY THANKS!!!! you just solved my issue
!
